### PR TITLE
Remove /cname endpoint

### PR DESCRIPTION
**Pull request by stsewd**
This isn't used or documented, it was added here https://github.com/rtfd/readthedocs.org/commit/3cada1dcf8ee5435aaa01a4899a16e44939743f1

This should be merged after #4730
Related https://github.com/rtfd/readthedocs.org/issues/1991#issuecomment-424875197

### CODE REVIEW

File: readthedocs/restapi/urls.py [hunk 1]
```diff
@@ -65,7 +65,6 @@
 function_urls = [
     url(r'embed/', core_views.embed, name='embed'),
```
Comment (stsewd) [inline readthedocs/restapi/urls.py]: This one isn't used either, and it doesn't work

File: readthedocs/core/utils/__init__.py [hunk 2]
```diff
@@ -61,14 +61,8 @@ def broadcast(type, task, args, kwargs=None, callback=None):  # pylint: disable=
...
 
-def clean_url(url):
-    parsed = urlparse(url)
-    if parsed.scheme or parsed.netloc:
-        return parsed.netloc
-    return parsed.path
-
-
 def cname_to_slug(host):
+    # TODO: remove
```
Comment (stsewd) [inline readthedocs/core/utils/__init__.py]: Need to remove this after #4730

File: readthedocs/restapi/urls.py (hunk 1 above)
Comment (ericholscher) [reply readthedocs/restapi/urls.py]: Yea, we should kill this. We have the `embed` in our `ext` repo that actually does work.

Comment (stsewd) [reply readthedocs/restapi/urls.py]: I opened https://github.com/rtfd/readthedocs.org/issues/4768

**Comment by stsewd** [comment]
I just wanted to do all the removal here p: there is nothing blocking this, I can do the removal in the other PR

**Pull request merged**